octopoes/octopoes/xtdb/client.py

```diff
@@ -59,11 +59,17 @@ def _verify_response(response: Response) -> None:
         try:
             response.raise_for_status()
         except HTTPStatusError as e:
+            logger.error(e.response.status_code)
             if e.response.status_code != codes.NOT_FOUND:
                 logger.error(response.request.url)
                 logger.error(response.request.content)
                 logger.error(response.text)
-            raise e
+                raise e
+            else:
+                if response.json()["error"] == "Node not found":
+                    raise NodeNotFound
```

Comment:
Raise should get an instance of the exception, not the class. It is also good to use `raise from` to make it clear that the exception is a consequence of the earlier exception.

```suggestion
                    raise NodeNotFound() from e
```